### DISCUSSION TITLE
Making all light icon colors more vibrant

### DISCUSSION THREAD

**Discussion by williamhCode** [kyazdani42/nvim-web-devicons#491] [Q&A]
I recently tried out a light theme, and noticed that the light variant colors are a lot more dull and dark. For example the blue and purple for the cpp/hpp-file icons just looks extremely washed out to me. Is there a way to make all the icon colors more vibrant for the light mode icons?

Also does anyone feel like the light color icons are all too dark and washed out? I know everyone's perception of color is different and it also varies depending on light conditions/displays, but for me the light variant colors just look way too dull compared to the dark variant colors.

<img width="345" alt="Screenshot 2024-09-08 at 4 46 47 PM" src="https://github.com/user-attachments/assets/abc87eec-f345-48b5-81c0-16d89f2b0ffb">
<img width="339" alt="Screenshot 2024-09-08 at 4 54 50 PM" src="https://github.com/user-attachments/assets/da0a179f-218c-496e-8b7d-1ebb4d1b469b">

**Comment by alex-courtis** [comment]
The dark icon colours are derived from the filetype project's colours, usually their icon.

The light icon colours are calculated using [a luminance formula](https://stackoverflow.com/questions/596216/formula-to-determine-perceived-brightness-of-rgb-color/596243#596243) (option 1)

The light icons must be visible on a white background.

Is there anything you can suggest to make this better for your background?

**Reply by williamhCode** [reply]
The luminance of the the icons are essentially reversed for the light themes if i'm understanding?
Although this makes total sense in terms of visibility and compatibility for all background colors, I'm wondering if there's another way of converting to look visually more similar to the dark icon colors.

For example, taking account of the bg color, and darkening until a certain contrast is achieved with the background. Or applying more darkening the more monochrome the icon color is (since most backgrounds are a white-ish color).

For my current theme at least, it seems like I can just use the dark icon colors, and darken a few icons like markdown, txt, and makefile. I think that adding an option to override the automatic color selection would be nice, as of right now I have to do this to get the dark icons in light mode.
```lua
vim.opt.background = "dark"
require("nvim-web-devicons").refresh()
vim.opt.background = "light"
```
<img width="347" alt="Screenshot 2024-09-09 at 10 44 30 AM" src="https://github.com/user-attachments/assets/b2b0b84a-b1da-48e9-a5a2-6e5706506dfc">

**Reply by alex-courtis** [reply]
> For my current theme at least, it seems like I can just use the dark icon colors, and darken a few icons like markdown, txt, and makefile.

I'm glad that works well.

> I think that adding an option to override the automatic color selection would be nice, as of right now I have to do this to get the dark icons in light mode.

Yes. Let's add an option, it sounds like it would be useful to others.

A setup option, say, `variant` = `light|dark` would be clearest. We could add a setter for variant, however YAGNI.

[Pull Requests](https://github.com/nvim-tree/nvim-web-devicons/blob/master/CONTRIBUTING.md) are most gratefully appreciated!
